force-app/main/default/classes/HouseholdNamingService.cls

```diff
@@ -163,12 +175,23 @@ public without sharing class HouseholdNamingService {
                     membersByHouseholdId.get(idFor(household)) == null ?
                             new List<Contact>() :
                             membersByHouseholdId.get(idFor(household));
-            if (settings.isAdvancedHouseholdNaming()) {
+            HouseholdName householdName = new HouseholdName(
```

Comment:
This adds complexity to the method. Can the inner loop logic, or part of it, be abstracted out into another method to be easier to read/debug/test?